### PR TITLE
[Merged by Bors] - proposal: sign proposal data like ballot

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
part of #4604

the minimum change required before 1st block

### CODE REVIEW

**Comment by countvonzero** [comment]
@dshulyak 

do you think we should sign Proposal in a different domain? currently it's signed with `signing.Ballot` domain.

**Comment by dshulyak** [comment]
it seems possible to construct malfeasance proof using proposal and ballot signature for same layer and it will be seen as valid. below is a valid malfeasance proof right?

proposal : layer || proposal hash || smesher || signature
ballot      : layer || ballot hash      || smesher || signature

because signature will be valid but msg hash is different. if they are in different domain there is no such problem

**Comment by countvonzero** [comment]
> it seems possible to construct malfeasance proof using proposal and ballot signature for same layer and it will be seen as valid. below is a valid malfeasance proof right?
> 
> proposal : layer || proposal hash || smesher || signature ballot : layer || ballot hash || smesher || signature
> 
> because signature will be valid but msg hash is different. if they are in different domain there is no such problem

yes. agreed. thank for confirming.

**Comment by countvonzero** [comment]
bors try

File: signing/signer.go [hunk 1]
```diff
@@ -18,9 +18,10 @@ type Domain byte
 const (
 	ATX Domain = 0
 
-	BALLOT = 2
-	HARE   = 3
-	POET   = 4
+	PROPOSAL = 1
```
Comment (dshulyak) [inline signing/signer.go:21]: looks strange that we didn't use 1 before

**Review by dshulyak — APPROVED** [review]

File: signing/signer.go (hunk 1 above)
Comment (countvonzero) [reply signing/signer.go:21]: @fasmat do you know why? 

maybe unrelated: 
staticcheck complains
`SA9004: only the first constant in this group has an explicit type (staticcheck)` 
if i don't leave a empty line.

Comment (dshulyak) [reply signing/signer.go:21]: i think we had beacon as 1, and then @fasmat implemented change to split it into different domains

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**